### ISSUE TITLE
Investigate retrieving CI status for uploads

### ISSUE THREAD

**Issue by tnt-sentry** [codecov/engineering-team#1493]
The UI already claims we do this (see parent issue) but this is actually representative of Codecov processing status. The CLI sends some amount of CI related info. We need to record what data we have and what data we can query for to better represent CI status in the UI. For GHA, we may even be able to use webhooks to get status updates. 

Outputs:
* Write up on what we currently receive from the CLI related to CI
* Write up on what data we can get from the CI providers with the given data
* * Note that this will be difficult for non GH if private
* What additional permissions/data if any would we need to accomplish this?

Focus primarily on GH -> CircleCI -> Bitrise.

**Comment by giovanni-guidini** [comment]
TL;DR
* We have a mechanism that queries the git provider for all the commit statuses, and we filter those to know if the CI passed or not (some statuses that we deem important, most are ignored)
* At least for GitHub we could also make better use of the webhook information we receive. This would help reduce the number of calls we make to the git provider.
* `Commit.ci_status` is a better reflection of wether a CI passed / failed for a commit

-----

The Notify task has a mechanism to check if CI was sucessful or not: https://github.com/codecov/worker/blob/d4496ed0e950f772c5ad0e33e770bf251223d87e/tasks/notify.py#L578

This is used to see if we should wait longer for customers that have `codecov.notify.wait_for_ci` YAML configured to true (default is true)


> 👀 **Read full investigation**     
>
> Full investigation [here](https://www.notion.so/sentry/CI-Statuses-f26aa1dda17b4fb08dd8a7d068d05ce4?pvs=4)
